plasmapy/physics/parameters.py

```diff
@@ -167,10 +170,18 @@ def Alfven_speed(B, density, ion="p"):
     if density.unit == units.m**-3:
         try:
             m_i = ion_mass(ion)
-            try:
-                Z = charge_state(ion)
-            except ValueError:
-                Z = 1
+            if z_mean == None:
+                warnings.warn("No z_mean given, defaulting to atomic charge",
+                              PhysicsWarning)
+                try:
+                    Z = charge_state(ion)
+                except ValueError:
+                    Z = 1
+            else:
+                # using average ionization provided by user
+                Z = z_mean
```

Comment:
I think this whole part from line 173 to here could be reasonably dumped into a _private function, to avoid duplication.

Reply:
That sounds like a terribly reasonable idea. Perhaps we can even call it something like `_boilerPlate()` 😛

Reply:
Try to be more descriptive, we might need multiple boiler plates :D

Reply:
How about `_ZBoilerPlate()`? With an emphasis on 'Zee', to be read like a mad German scientist.